pkg/avatar/initials_convert_test.go

```diff
@@ -39,3 +40,32 @@ func TestInitialsPNG(t *testing.T) {
 
 	require.Equal(t, expectImg.Bounds(), resImg.Bounds(), "images doesn't have the same size")
 }
+
+func Test_Initials_PNG_Shutdown(t *testing.T) {
+	if testing.Short() {
+		t.Skipf("this test require the \"convert\" binary, skip it due to the \"--short\" flag")
+	}
+
+	client, err := NewPNGInitials("convert")
+	require.NoError(t, err)
+
+	var rawRes []byte
+	generateFinished := false
+	isStarted := make(chan struct{}, 1)
+	go func() {
+		isStarted <- struct{}{}
+		rawRes, err = client.Generate(context.Background(), "JD", "#FF7F1B")
+		generateFinished = true
+	}()
+
+	// Ensure that the Generate is started and not finished before calling
+	// `Shutdown`
+	<-isStarted
+	require.False(t, generateFinished)
```

Comment:
I don't like this test: it can easily be flaky. On this line, we don't have any guarantee that `Generate` is started. The call to `client.Shutdown` may take the lock before the call to `client.Generate`.

Reply:
I can add a little sleep to be sure that `Generate` have the time to start but except this I don't really see how to improve this.

Reply:
I'm still not happy with this test. It could be flaky even with this change, and `go test -race` is also red (for the same reason).

Reply:
I don't really see how to test it so I will remove the test